src/tiles.jl

```diff
@@ -107,6 +107,26 @@ function Tile(point, zoom::Integer, crs::WGS84)
     return Tile(xtile, ytile, zoom)
 end
 
+"""
+    z_index(extent::Extent, res::NamedTuple, crs::WebMercator) => Int
+
+Calculate a z value from `extent` and pixel resolution `res` for `crs`.
+The rounded mean calculated z-index for X and Y resolutions is returned.
+
+`res` should be `(X=xres, Y=yres)` to match the extent.
+
+We assume tiles are the standard 256*256 pixels. Note that this is not an
+enforced standard, and that retina tiles are 512*512.
+"""
+function z_index(extent::Extent, res::NamedTuple, crs::WebMercator)
```

Comment:
Should this also accept tuples (x, y) as Makie does?  In that case we can also make `z_index(..., Tuple) = z_index(..., (; X = ..., Y = ...))` or so.

Reply:
There should probably also be a catchall method to error when `crs` is not `WebMercator` (for now...)